pkg/api/customization/cluster/actionhandler.go

```diff
@@ -684,18 +684,39 @@ func (a ActionHandler) RestoreFromEtcdBackupHandler(actionName string, action *t
 	// checking access
 	var mgmtCluster mgmtclient.Cluster
 	if err := access.ByID(apiContext, apiContext.Version, apiContext.Type, apiContext.ID, &mgmtCluster); err != nil {
-		response["message"] = "none existent Cluster"
+		response["message"] = "nonexistent Cluster"
 		apiContext.WriteResponse(http.StatusBadRequest, response)
 		return errors.Wrapf(err, "failed to get Cluster by ID %s", apiContext.ID)
 	}
 
 	cluster, err := a.ClusterClient.Get(apiContext.ID, metav1.GetOptions{})
 	if err != nil {
-		response["message"] = "none existent Cluster"
+		response["message"] = "nonexistent Cluster"
 		apiContext.WriteResponse(http.StatusBadRequest, response)
 		return errors.Wrapf(err, "failed to get Cluster by ID %s", apiContext.ID)
 	}
 
+	clusterBackupConfig := cluster.Spec.RancherKubernetesEngineConfig.Services.Etcd.BackupConfig
```

Comment:
I think we disable restore action for non-rke clusters, so just to confirm that assuming rke config not being null is safe here.

Reply:
Yeah I think it should be safe, action is added only for rke clusters with backup config enabled